packages/middleware/src/apiClientFactory/index.ts

```diff
@@ -102,6 +97,26 @@ const apiClientFactory = <
 
       const isApiFactory = typeof apiOrApiFactory === "function";
       const api = isApiFactory ? apiOrApiFactory(settings) : apiOrApiFactory;
+
+      const namespacedExtensions: Record<string, any> = {};
+      let sharedExtensions = customApi;
+
+      // If the extension is namespaced, we need to merge the extended api methods into the namespace
+      // Otherwise, we can just merge the extended api methods into the api
+      rawExtensions.forEach((extension) => {
+        if (extension.isNamespaced) {
+          namespacedExtensions[extension.name] = {
+            ...(namespacedExtensions?.[extension.name] ?? {}),
+            ...extension.extendApiMethods,
+          };
+        } else {
+          sharedExtensions = {
```

Comment:
Same here:

```ts
Object.assign(sharedExtensions, extension.extendApiMethods)
```

And then you could declare `sharedExtensions` with `const` instead of `let`.

Reply:
Still it is not possible because I do something like `sharedExtensions = Object.assign(sharedExtensions, blabla)` and the assignment is not allowed if a variable is a const.

Reply:
But I meant not doing the assignment because `Object.assign()` will already modify the object

Reply:
```js
sharedExtensions = Object.assign(sharedExtensions, blabla)
// equals
Object.assign(sharedExtensions, blabla)
```